### ISSUE TITLE
Segfault using sentencepiece with tf-nightly

### ISSUE THREAD

**Issue by mattdangerw** [google/sentencepiece#813]
We recently hit an [odd issue](https://github.com/keras-team/keras-nlp/issues/694) on our `tf-nightly` testing, where importing tensorflow first, then sentencepiece and running the sentencepiece trainer will cause a segfault.

The segfault comes from [GetRandomGeneratorSeed](https://github.com/google/sentencepiece/blob/631420b84be518c907060fd947aac01762d7fbb0/src/util.cc#L33). I cannot reproduce this in a colab sadly, looking at the code, I wonder if this has something to with `FLAGS_random_seed` that perhaps won't get trigger in a colab environment?

Steps to reproduce

- Setup a tf-nightly install. (Showing conda here but have repro'd without as well)
```sh
conda create --name sentencepiece-test python=3.9
conda activate sentencepiece-test
pip install tf-nightly sentencepiece
```
- Run the following.
```python
import io
# Failure only if tensorflow import is first!
import tensorflow as tf
import sentencepiece

bytes_io = io.BytesIO()
vocab_data = tf.data.Dataset.from_tensor_slices(
    ["the quick brown fox", "jumped over the dog"]
)
sentencepiece.SentencePieceTrainer.train(
    sentence_iterator=vocab_data.as_numpy_iterator(),
    model_writer=bytes_io,
    vocab_size=10,
    model_type="WORD",
)
```
- Segfault! Here's a log from our [testing ci](https://github.com/keras-team/keras-nlp/actions/runs/4010830202/jobs/6887761654). When I ran the failure through `gdb` the backtrace pointed to `GetRandomGeneratorSeed`.

**Comment by taku910** [comment]
Thank you for the report. Reproduced this error on my environment.

**Comment by taku910** [comment]
One workaround is to specify the random seed value just before importing sentencepece.

```
import tensorflow as tf
import sentencepiece

sentencepiece.SetRandomGeneratorSeed(0)
```

**Comment by taku910** [comment]
FYI: https://github.com/sass/libsass/issues/3151

**Comment by taku910** [comment]
Will fix it in the next release.

**Comment by brianroland** [comment]
@taku910 : jfyi, I see a similar/same segfault, while running official/nlp/data/train_sentencepiece.py from tf-models-official, with the wmt14_translate/de-en dataset from tensorflow datasets.  Per your earlier comment, if I put sentencepiece.SetRandomGeneratorSeed(0) immediately after import, I no longer see the segfault, and execution proceeds with sensible output.  Thanks for your help.  Curious, what's the release cadence?

Python 3.10.9
sentencepiece==0.1.97
tensorflow==2.12.0
tensorflow-datasets==4.9.0
tf-models-official==2.12.0

**Comment by taku910** [comment]
Fixed in v0.1.98

**Comment by brianroland** [comment]
Thanks!
